executor/join.go

```diff
@@ -337,14 +314,32 @@ func (e *HashJoinExec) initializeForProbe() {
 func (e *HashJoinExec) fetchOuterAndProbeHashTable(ctx context.Context) {
 	e.initializeForProbe()
 	e.workerWaitGroup.Add(1)
-	go e.fetchOuterChunks(ctx)
+	go util.WithRecovery(func() { e.fetchOuterChunks(ctx) }, e.disposeOuterFetcher)
 
 	// Start e.concurrency join workers to probe hash table and join inner and outer rows.
 	for i := uint(0); i < e.concurrency; i++ {
 		e.workerWaitGroup.Add(1)
-		go e.runJoinWorker(i)
+		workID := i
```

Comment:
remove this inline and:
```go
go util.WithRecovery(func() { e.runJoinWorker(i) }, e.disposeJoinWorker)
```

Reply:
wuu... this is a hole cased by we use closure in for-loop.

https://play.golang.org/p/XbUXo-qlb3I